cmd/xl-v1-object.go

```diff
@@ -348,22 +329,22 @@ func (xl xlObjects) getObject(ctx context.Context, bucket, object string, startO
 			partLength = length - totalBytesRead
 		}
 
+		endOffset := erasure.ShardFileEndOffset(partOffset, partLength, partSize)
 		// Get the checksums of the current part.
-		bitrotReaders := make([]*bitrotReader, len(onlineDisks))
+		readers := make([]io.ReaderAt, len(onlineDisks))
 		for index, disk := range onlineDisks {
 			if disk == OfflineDisk {
 				continue
 			}
 			checksumInfo := metaArr[index].Erasure.GetChecksumInfo(partName)
-			endOffset := getErasureShardFileEndOffset(partOffset, partLength, partSize, xlMeta.Erasure.BlockSize, xlMeta.Erasure.DataBlocks)
-			bitrotReaders[index] = newBitrotReader(disk, bucket, pathJoin(object, partName), checksumInfo.Algorithm, endOffset, checksumInfo.Hash)
+			readers[index] = newBitrotReader(disk, bucket, pathJoin(object, partName), endOffset, checksumInfo.Algorithm, checksumInfo.Hash, erasure.ShardSize())
 		}
-
-		err := erasure.Decode(ctx, writer, bitrotReaders, partOffset, partLength, partSize)
+		err := erasure.Decode(ctx, writer, readers, partOffset, partLength, partSize)
+		closeBitrotReaders(readers)
```

Comment:
This should be in defer or no?

Reply:
Better close it here since this is inside the loop `for ; partIndex <= lastPartIndex; partIndex++ { }` i.e if there are 100 parts and 16 disks, 1600 connections will be open by the time GetObject is done

Reply:
That is not very evident in the code, can you explain with a comment?

Reply:
Comment is needed on why we close readers here..